test/test-metrics.c

```diff
@@ -0,0 +1,159 @@
+#include "uv.h"
+#include "task.h"
+#include <string.h> /* memset */
+
+
+typedef struct {
+  int metrics_run_count;
+  int timer_run_count;
+} mtc_t;
+
+typedef struct {
+  uv_async_t async;
+  uv_loop_t mloop;
+  uv_timer_t timer;
+  uv_thread_t thread;
+  uv_timer_t ttimer;
+  uint64_t thread_timer_count;
+} mthread_t;
+
+
+static void timer_noop_cb(uv_timer_t* handle) {
+  (*(int*) handle->data)++;
+}
+
+
+TEST_IMPL(metrics_idle_time) {
+  uv_timer_t timer;
+  uint64_t pt;
+  int cntr;
+  int r;
+
+  cntr = 0;
+  timer.data = &cntr;
+  uv_loop_configure(uv_default_loop(), UV_METRICS_IDLE_TIME);
+  uv_timer_init(uv_default_loop(), &timer);
+
+  uv_timer_start(&timer, timer_noop_cb, 100, 0);
+
```

Comment:
Nit: check return values of the uv_ methods here and in the other tests

Reply:
Fixed them all.